client/src/components/Survey.js

```diff
@@ -2,82 +2,88 @@ import React, {useEffect, useState} from "react";
 import {Button, Form, Select, RangeSlider, FormLayout, TextField} from '@shopify/polaris';
 
 export function Survey({ survey = {} }) {
-    const [answers, setAnswers] = useState({});
+    const [questions, setQuestions] = useState([]);
 
     useEffect(() => {
-        if(survey.questions) {
-            survey.questions.forEach((question) => {
-                answers[question.question] = null
-                setAnswers(answers)
-                console.log('useeffect')
-            })
-        }
-        // eslint-disable-next-line
+      let qs = survey.questions || []
+      setQuestions(qs.map(q => {
+        q.answers = []
+        return q
+      }))
     }, [survey])
 
-    const questionMarkup = (question) => {
-        if(question.min){
-            // Range question
+    async function submitAnswers() {
+      const res = await fetch(`/api/answers`, {
+        method: "POST",
+        headers: {
+          "Accept": "application/json",
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify(questions)
+      });
+
+      await res
+        .json()
+        .catch(err => console.log(err));
+    }
+
+    const questionMarkup = (question, index) => {
+        switch(question.type) {
+          case "range":
```

Comment:
we should probably define these types somewhere as enums/constants

Reply:
we haven't used in another part of the client yet, so we should probably wait for another pr that does. I also don't know a good pattern for where to place constants in js lol